### PR TITLE
refactor: simplify bundler plugin registration

### PR DESCRIPTION
**Pull request by BobbieGoede**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [x] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description
Reduces duplicate options and uses `addBuildPlugin` where possible, inspired from Nuxt's source code.

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have added tests (if possible).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/bundler.ts [hunk 1]
```diff
@@ -67,35 +61,29 @@ export async function extendBundler({ options: nuxtOptions }: I18nNuxtContext, n
...
-  addVitePlugin(TransformMacroPlugin.vite(sourceMapOptions))
-  addVitePlugin(ResourcePlugin.vite(sourceMapOptions))
-  if (nuxtOptions.experimental.autoImportTranslationFunctions) {
-    addVitePlugin(TransformI18nFunctionPlugin.vite(sourceMapOptions))
-  }
-
+  addVitePlugin(
+    VueI18nVitePlugin({
+      ...vueI18nPluginOptions,
+      fullInstall: nuxtOptions.bundle.fullInstall,
```
Comment (BobbieGoede) [inline src/bundler.ts]: @kazupon 
These options were not set for the webpack plugin, was this intentional? 🤔

Comment (kazupon) [reply src/bundler.ts]: @BobbieGoede 
Ah, I had forgotten these options for webpack. 😅 
you can specify these options for webpack.

File: package.json [hunk 2]
```diff
@@ -81,7 +81,8 @@
     "overrides": {
       "@nuxtjs/i18n": "link:.",
       "consola": "^3",
-      "unimport": "3.13.4"
+      "unimport": "3.13.4",
+      "rollup": "4.28.1"
```
Comment (kazupon) [inline package.json:86]: why do we need to override? 🤔

Comment (BobbieGoede) [reply package.json:86]: This is to resolve the type mismatch for the nitro plugins like [`acfc08f` (#3273)](https://github.com/nuxt-modules/i18n/pull/3273/commits/acfc08f1296b5f55cc8424786160f1131c6f7f9f#diff-126f9d42f327404ca57cb5a71d7c101fe1b0ee34dc04757b61cf37d09b21f257L64), I was wondering how Nuxt isn't running into the same issue and found they also override the resolution for rollup https://github.com/nuxt/nuxt/blob/main/package.json#L59

File: src/bundler.ts (hunk 1 above)
Comment (BobbieGoede) [reply src/bundler.ts]: I see! changed it mow so all options are the same for vite and webpack 💪

**Pull request merged**